### PR TITLE
pipeline(ticdc): reduce go routine size

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref #4351

### What is changed and how it works?
remove 4n gotoutines 
1. synctable account metrics 
2. flowcontroller abort
3. puller metrics
4. MonitorCancelLatency goroutine

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
None`.
```

### CODE REVIEW

File: cdc/processor/pipeline/sorter.go [hunk 1]
```diff
@@ -160,6 +152,10 @@ func (n *sorterNode) StartActorNode(ctx pipeline.NodeContext, isTableActorMode b
 		for {
 			select {
 			case <-stdCtx.Done():
+				// Since the flowController is implemented by `Cond`, it is not cancelable
+				// by a context. We need to listen on cancellation and aborts the flowController
+				// manually.
+				n.flowController.Abort()
```
Comment (liuzix) [inline cdc/processor/pipeline/sorter.go:158]: The current implementation of `flowController` cannot be canceled by contexts, so the `Abort()` may never be called, causing a deadlock.

**Review by liuzix — CHANGES_REQUESTED** [review]

File: cdc/processor/pipeline/sorter.go (hunk 1 above)
Comment (sdojjy) [reply cdc/processor/pipeline/sorter.go]: fixed

File: cdc/processor/processor.go [hunk 2]
```diff
@@ -964,17 +965,6 @@ func (p *processor) createTablePipelineImpl(ctx cdcContext.Context, tableID mode
 		sink,
 		p.changefeed.Info.GetTargetTs(),
 	)
-	p.wg.Add(1)
-	p.metricSyncTableNumGauge.Inc()
-	go func() {
-		table.Wait()
-		p.wg.Done()
```
Comment (amyangfei) [inline cdc/processor/processor.go:971]: Can we ensure table pipeline has exited after removing these lines

Comment (sdojjy) [reply cdc/processor/processor.go:971]: yes, we call `table.Wait()` after we remove a table

File: cdc/processor/pipeline/sorter.go [hunk 3]
```diff
@@ -315,6 +307,7 @@ func (n *sorterNode) Destroy(ctx pipeline.NodeContext) error {
 			log.Warn("schedule table cleanup task failed", zap.Error(err))
 		}
 	}
+	n.flowController.Abort()
```
Comment (overvenus) [inline cdc/processor/pipeline/sorter.go:310]: Please add some comments about deadlocking.

File: pkg/util/cancel_monitor.go [hunk 4]
```diff
@@ -22,11 +22,10 @@ import (
 )
 
 // MonitorCancelLatency monitors the latency from ctx being cancelled and the returned function being called
-func MonitorCancelLatency(ctx context.Context, identifier string) func() {
+func MonitorCancelLatency(ctx context.Context, identifier string) (func(), func()) {
```
Comment (overvenus) [inline pkg/util/cancel_monitor.go:25]: Could you add some comments about the returned closures?

File: cdc/sorter/unified/unified_sorter.go [hunk 5]
```diff
@@ -114,7 +114,7 @@ func (s *Sorter) Run(ctx context.Context) error {
 
 	defer close(s.closeCh)
 
-	finish := util.MonitorCancelLatency(ctx, "Unified Sorter")
+	startCancel, finish := util.MonitorCancelLatency(ctx, "Unified Sorter")
```
Comment (overvenus) [inline cdc/sorter/unified/unified_sorter.go:117]:
```suggestion
	finish, startCancel := util.MonitorCancelLatency(ctx, "Unified Sorter")
```

Putting cancel in seconds position is more idiomatic.

File: cdc/processor/pipeline/sorter.go (hunk 3 above)
Comment (sdojjy) [reply cdc/processor/pipeline/sorter.go:310]: done

File: pkg/util/cancel_monitor.go (hunk 4 above)
Comment (sdojjy) [reply pkg/util/cancel_monitor.go:25]: done

File: cdc/sorter/unified/unified_sorter.go (hunk 5 above)
Comment (sdojjy) [reply cdc/sorter/unified/unified_sorter.go:117]: done

**Review by overvenus — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**